### PR TITLE
Country Code

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
Adds a call to our internal api endpoint `/geolocation` to get the country that we think a user is in based on their IP address, stores the result in the local state of `App` and remembers it for 10 days in local storage.

`App` is our top level component making this state effectively global.

### Why are we using localStorage and not a cookie?
Because we don't need to pass this back to the server and localStorage is 🚤 🐎 🚀 and cookies are 🐌 🐢 👎 

We save this data as:

```typescript
type LocalCountryCodeType = {
    value: string;
    expires: number;
};
```

### What about how frontend stores this value?
FE seems (admittedly, I just clicked around a bit watching dev tools) to only call this api from within `discussion` and it also uses local state. I used the same key name so values will live between applications.

## Why?
Because it's a useful value to know in general but specifically because Slot Machine needs to know this and doesn't want to make the same call multiple times from each of their slots.

## Link to supporting Trello card
https://trello.com/c/kBAFb6aK/1115-implement-country-code-in-client-side-apps

### CODE REVIEW

File: src/web/browser/App.tsx [hunk 1]
```diff
@@ -42,11 +44,23 @@ export const hydrateApp = ({ CAPI, NAV }: { CAPI: CAPIType; NAV: NavType }) => {
...
     useEffect(() => {
         setIsSignedIn(!!getCookie('GU_U'));
     }, []);
 
+    useEffect(() => {
+        const callFetch = async () => {
+            setCountryCode(await getCountryCode());
+        };
+        callFetch();
+    }, []);
```
Comment (liywjl) [inline src/web/browser/App.tsx]: I think this is a little clearer, but again just my opinion
```
useEffect(async () => {
    try {
        const countryCodeResponse = await getCountryCode()
        setCountryCode(countryCodeResponse)
    } catch (e) {
        // log some error
    }
}, []);
```

File: src/web/browser/App.tsx [hunk 2]
```diff
@@ -42,11 +44,23 @@ export const hydrateApp = ({ CAPI, NAV }: { CAPI: CAPIType; NAV: NavType }) => {
...
+
+    console.log(
```
Comment (liywjl) [inline src/web/browser/App.tsx]: console.log

File: src/web/lib/getCountryCode.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                    value: countryCode,
+                    expires: new Date().getTime() + TEN_DAYS,
+                }),
+            );
+        });
+        // Return the country value that we got from our fetch call
+        return countryCode;
+    }
+    // THere was no need to fetch, return the local value
+    return localCountryCode.value;
```
Comment (liywjl) [inline src/web/lib/getCountryCode.ts]: prob best to wrap this in an `else` block, it makes it easier to reason with

File: src/web/browser/App.tsx (hunk 2 above)
Comment (oliverlloyd) [reply src/web/browser/App.tsx]: That one was intentional! I wanted to give Andre a pointer but also keep typescript happy (no unused vars...)

File: src/web/browser/App.tsx (hunk 1 above)
Comment (oliverlloyd) [reply src/web/browser/App.tsx]: As per [this great article](https://www.robinwieruch.de/react-hooks-fetch-data), useEffect can't return a promise so the function has to be defined inside

File: src/web/lib/getCountryCode.ts (hunk 3 above)
Comment (oliverlloyd) [reply src/web/lib/getCountryCode.ts]: I did. Our linter ripped it away and replaced it with what you see. 🤷‍♂ Prettier is God

**Pull request merged**